hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/HssCallAttempt.java

```diff
@@ -149,4 +155,47 @@ public boolean isScheduleRedirect() {
         }
         return null;
     }
+
+    /**
+     * Extracts the key set for scheduled calls.
+     *
+     * @return the key set
+     */
+    public Set<Key> keySetFor() {
+        final var sender = nativeOperations().getAccount(senderId());
+        requireNonNull(sender);
+        if (sender.smartContract()) {
+            return getKeysForContractSender();
+        } else {
+            return getKeysForEOASender();
+        }
+    }
+
+    @NonNull
+    private Set<Key> getKeysForEOASender() {
+        // If an Eth sender key is present, use it. Otherwise, use the account key if present.
```

Comment:
Comment is inconsistent with implementation; should be,
```
// For a top-level EthereumTransaction, use the Ethereum sender key; otherwise, 
// use the full set of simple keys authorizing the ContractCall dispatching this 
// HSS call attempt
```